From 5708c7e9804886c2115e759aad79c6da0edafcba Mon Sep 17 00:00:00 2001
From: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Date: Tue, 9 Oct 2012 11:33:25 +0200
Subject: [PORT FROM MAIN] Mismatched free()/delete/delete[]

BZ: 61157

The purpose of this patch is to fix the (invalid) use of delete
instead of delete[] in audioflinger, line 6919 and line 6921.
This error has been found by valgrind.

Feature: None
Category: Improvement
Purpose: AOSP+
Origin: Internal
Upstream-candidate: yes
Change-Id: I9dcc7458971afa3be0d5b2b495e87b71c01998a9
Orig-Change-Id: Ifd2e35519e1daaa3910d423153ade0441a608008
Signed-off-by: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Reviewed-on: http://android.intel.com:8080/82122
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 services/audioflinger/Threads.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 0d8b137..b028510 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -4338,9 +4338,9 @@ void AudioFlinger::RecordThread::audioConfigChanged_l(int event, int param) {
 
 void AudioFlinger::RecordThread::readInputParameters()
 {
-    delete mRsmpInBuffer;
+    delete[] mRsmpInBuffer;
     // mRsmpInBuffer is always assigned a new[] below
-    delete mRsmpOutBuffer;
+    delete[] mRsmpOutBuffer;
     mRsmpOutBuffer = NULL;
     delete mResampler;
     mResampler = NULL;
-- 
1.7.4.1

